### PR TITLE
Update APIs to match Lattigo branch

### PR DESCRIPTION
**Pull request by crockeea**
*Issue #, if available:*

*Description of changes:*
This PR adds all of the recent changes made to the lattigo branch to the SEAL branch. This includes:
- Splitting `DepthFinder` into `ExplicitDepthFinder` and `ImplicitDepthFinder`
- Adding the `RotationSet` evaluator
- Adding `CKKSParams` API and updating `HEContext` API

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/hit/api/evaluator/explicitdepthfinder.cpp [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    CKKSCiphertext ExplicitDepthFinder::encrypt(const vector<double> &) {
+        LOG_AND_THROW_STREAM("ExplicitDepthFinder does not define encrypt() with an implicit level");
+    }
+
+    CKKSCiphertext ExplicitDepthFinder::encrypt(const vector<double> &, int level) {
+        if (level < 0) {
+            LOG_AND_THROW_STREAM("Explicit encryption level must be non-negative, got " << level);
+        }
+
+        max_contiguous_depth = max(max_contiguous_depth, level);
```
[the remark this answers is not in the document]
Comment (crockeea) [reply src/hit/api/evaluator/explicitdepthfinder.cpp:23]: Good catch. Fixed.

File: src/hit/api/evaluator/explicitdepthfinder.h [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+        int num_slots() const override;
+
+       protected:
+        void rescale_to_next_inplace_internal(CKKSCiphertext &ct) override;
+
+       private:
+        const int num_slots_ = 4096;
+        // We can't make this value `const` even though ExplicitDepthFinder
+        // doesn't update it. The reason is that ExplicitDepthFinder works when
```
[the remark this answers is not in the document]
Comment (crockeea) [reply src/hit/api/evaluator/explicitdepthfinder.h:59]: Old comment; removed.

File: src/hit/api/evaluator/implicitdepthfinder.h [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+        int num_slots() const override;
+
+       protected:
+        void rescale_to_next_inplace_internal(CKKSCiphertext &ct) override;
+
+       private:
+        const int num_slots_ = 4096;
+        // We can't make this value `const` even though ImplicitDepthFinder
+        // doesn't update it. The reason is that ImplicitDepthFinder works when
```
[the remark this answers is not in the document]
Comment (crockeea) [reply src/hit/api/evaluator/implicitdepthfinder.h:58]: Old comment; removed.

**Review by SalusaSecondus — APPROVED** [review]
LGTM

**Pull request merged**